resources/js/components/Teams/TeamMap.vue

```diff
@@ -1,109 +1,69 @@
             </button>
-
-            <l-map :zoom="zoom" :center="center" :minZoom="1">
-                <l-tile-layer :url="url" :attribution="attribution"/>
-                <v-marker-cluster v-if="geojson.length">
-                    <l-marker v-for="i in geojson" :lat-lng="i.properties.latlng" :key="i.properties.id">
-                        <l-popup :content="content(i)" :options="options"/>
-                    </l-marker>
-                </v-marker-cluster>
-            </l-map>
+            <cluster-map
+                :clusters-url="`/teams/clusters/${teamId}`"
+                :points-url="`/teams/points/${teamId}`"
+                @loading-complete="loading = false"
+            />
         </fullscreen>
     </div>
 </template>
 
 <script>
-import { LMap, LTileLayer, LMarker, LPopup } from 'vue2-leaflet';
-import Vue2LeafletMarkerCluster from 'vue2-leaflet-markercluster';
-import {mapHelper} from '../../maps/mapHelpers';
+import Loading from 'vue-loading-overlay';
+import 'vue-loading-overlay/dist/vue-loading.css';
+import ClusterMap from '../../views/global/ClusterMap';
 
 export default {
     name: 'TeamMap',
+    props: ['teamId'],
     components: {
-        LMap,
-        LTileLayer,
-        LMarker,
-        LPopup,
-        'v-marker-cluster': Vue2LeafletMarkerCluster
-    },
-    created ()
-    {
-        this.attribution += new Date().getFullYear();
+        Loading,
+        ClusterMap
     },
-    data ()
-    {
+    data() {
         return {
-            zoom: 2,
-            center: L.latLng(0,0),
-            url:'https://{s}.tile.osm.org/{z}/{x}/{y}.png',
-            attribution:'Map Data &copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors, Litter data &copy OpenLitterMap & Contributors ',
-            loading: true,
-            fullscreen: false,
-            options: mapHelper.popupOptions
-        };
+            loading: true
+        }
     },
-    computed: {
-
-        /**
-         * From backend api request
-         */
-        geojson ()
+    watch: {
+        teamId (id)
         {
-            return this.$store.state.teams.geojson
-                ? this.$store.state.teams.geojson.features
-                : [];
+            if (id > 0) this.loading = true;
         }
     },
     methods: {
-
-        /**
-         * Return html content for each popup
-         *
-         * Translate tags
-         *
-         * Format datetime (time image was taken)
-         */
-        content (feature)
-        {
-            return mapHelper.getMapImagePopupContent(
-                feature.properties.img,
-                feature.properties.text,
-                feature.properties.datetime,
-                feature.properties.picked_up,
-                '',
-                ''
-            );
-        },
-
         fullscreenChange (fullscreen)
         {
-            this.fullscreen = fullscreen
+            this.fullscreen = fullscreen;
         },
 
         toggle ()
```

Comment:
Comments needed on every function and computed property

Reply:
This code is basically copy-pasted from the User profile map. 👍